src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -358,7 +360,10 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
     s.asInstanceOf[S with EfficientSplit]
   }
 
-  private[this] def growTable(newlen: Int) = {
+  private[this] def growTable(newlenLong: Long) = {
+    if (newlenLong > Int.MaxValue)
+      throw new BufferOverflowException()
+    val newlen = newlenLong.toInt
     var oldlen = table.length
     threshold = newThreshold(newlen)
```

Comment:
while this PR does address the overflow problem (and I think that is sufficiently valuable to move this PR forward), it still does not address one of the concerns in the linked issue - namely that we compute some `threshold` value here but never use it. I think it is important to investigate and determine the original intention of computing `threshold`/calling `newThreshold`, and clean up the method accordingly.

consequently, I don't think we should say that this PR fixes the issue (especially because GitHub will then automatically close the issue), but rather that it partially addresses the issue.

Reply:
I've fixed the comment which said that this PR fixes the issue. I'll try to address the problem with `threshold` separately.

Reply:
Huh?  I don't understand the comment.  `threshold` is used all over the code to determine when it's time to grow again.

You might want to special-case it for being unable to grow (e.g. make it bigger), but it's used in every method that creates a new entry.

Reply:
my mistake. I missed the fact that it's not a local `var`/`val`